src/custom-elements/src/components/textarea/textarea.tsx

```diff
@@ -0,0 +1,122 @@
+import { Component, Element, Prop, Watch, h } from '@stencil/core';
+import { CHI_STATES, ChiStates } from '../../constants/states';
+import { ICON_COLORS, IconColors } from '../../constants/color';
+import { TEXT_INPUT_SIZES, TextInputSizes } from '../../constants/size';
+
+@Component({
+  tag: 'chi-textarea',
+  styleUrl: 'textarea.scss',
+  scoped: true
+})
+
+export class Textarea {
+  @Element() el: HTMLChiTextareaElement;
+  /**
+   * To define state color of Textarea
+   */
+  @Prop({ reflect: true }) state: ChiStates;
+  /**
+   * To add a left positioned icon
+   */
+  @Prop({ reflect: true }) iconLeft: string;
+  /**
+   * To add a right positioned icon
+   */
+  @Prop({ reflect: true }) iconRight: string;
+  /**
+   * To define color of left icon
+   */
+  @Prop({ reflect: true }) iconLeftColor: IconColors;
+  /**
+   * To define color of right icon
+   */
+  @Prop({ reflect: true }) iconRightColor: IconColors;
+  /**
+   * To define size of Textarea
+   */
+  @Prop({ reflect: true }) size ?: TextInputSizes = 'md';
+  /**
+   * To define name of Textarea
+   */
+  @Prop({ reflect: true }) name: string;
+  /**
+   * To disable Textarea
+   */
+  @Prop({ reflect: true }) disabled = false;
+  /**
+   * To define -hover, -focus statuses
+   */
+  @Prop() _status: string;
```

Comment:
```suggestion
  @Prop() _status: '-hover' | '-focus';
```

Reply:
Thanks for suggestions Dani.
I will add the `onChiChange`, `onChiInput`, `onChiFocus`, `onChiBlur` events.

Regarding value prop: I used `<slot>` logic instead of value Prop because native Textarea doesn't have Value attribute and also, unlike text input element, it supports line breaks within text. With this approach you can still access `chi-textarea` value with:
`.textContent`. What do you think about it?

Reply:
Using slot is perfect, but you'll need to update a value prop on the component. textarea doesn't have the `value` attribute but do have the `value` prop. So, just don't set the `reflect` option in the decorator.  Remember attribute is HTML, prop is javascript. You can access it `document.querySelector("textarea").value`

Reply:
All done, thanks!